Kernel/Interrupts/IRQHandler.cpp

```diff
@@ -11,11 +11,12 @@
 
 namespace Kernel {
 
-IRQHandler::IRQHandler(u8 irq)
+IRQHandler::IRQHandler(u8 irq, bool setup_irq)
     : GenericInterruptHandler(irq)
     , m_responsible_irq_controller(InterruptManagement::the().get_responsible_irq_controller(irq))
 {
-    disable_irq();
+    if (setup_irq)
```

Comment:
would it be more clear to call this `should_disable_irq` instead?

Reply:
probably not - the most important thing disable_irq() does in this case is overwriting the irq handler